src/Type/TypeCombinator.php

```diff
@@ -776,15 +778,17 @@ public static function intersect(Type ...$types): Type
 
 			return $union;
 		}
+		$typesCount = count($types);
 
 		// transform A & (B & C) to A & B & C
-		for ($i = 0; $i < count($types); $i++) {
+		for ($i = 0; $i < $typesCount; $i++) {
```

Comment:
`count(array)` is O(1) in PHP, I belive this can not provide measurable performance improvement

Reply:
previously it was called in every loop iteration which would make it O(n) then I guess? AFAIK this one of the micro optimizations that also the EA Extended plugin for phpstorm suggests and it definitely makes sense IMO.

Reply:
Although the complexity of count() is O(1), storing the result of count() in a variable is much faster than calling them again.